src/redux/selectors/index.js

```diff
@@ -1,6 +1,34 @@
-export const getSearchDetails = state => state.imagesGalleryPage
+import { createSelector } from 'reselect'
+
+import { ITEMS_PER_PAGES } from 'common/constants'
+
+// General
+export const getIsLoading = state => state.imgDailyPage.isLoading
+
+//  Image gallery
+export const getImages = state => state.imagesGalleryPage.items
+
+export const getImagesSelector = createSelector(
+  getImages,
+  items => items.flatMap(({ links }) => links),
+)
 
+// Daily image
+export const getDailyImage = state => state.imgDailyPage.image
+
+// Search
+export const getSearchDetails = state => state.imagesGalleryPage
 export const getSearchText = state => state.imagesGalleryPage.searchText
+
+// Pagination
 export const getCurrentPage = state => state.imagesGalleryPage.pageIndex
+export const getTotalItemsOnPage = state => state.imagesGalleryPage.totalItems
```

Comment:
there are some repetitions here, for instance, this has **imagesGalleryPage** repeated and destructured 2 times:
```
export const getTotalItemsOnPage = state => state.imagesGalleryPage.totalItems
export const getSearchText = state => state.imagesGalleryPage.searchText
```
and you already have above one selector called:
```
export const getSearchDetails = state => state.imagesGalleryPage
```
the idea is that you should try to reuse this **getSearchDetails** in the above 2 - to avoid destructuring again and again.
(also the name is not correct I think, if your selector is named: getSearchDetails but it gets you state.imagesGalleryPage :) it should be named something like **getImagesGalleryPage** or **imagesGalleryPageSelector)**
```
export const getImagesGalleryPage = state => state.imagesGalleryPage

export const getTotalItemsOnPage = createSelector(
  getImagesGalleryPage, // right here we are reusing -> and we're not destructuring it from store again 
  imagesGalleryPage => imagesGalleryPage.totalItems
)

export const getSearchText= createSelector(
  getImagesGalleryPage,
  imagesGalleryPage => imagesGalleryPage.searchText
)
```